Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -150,11 +148,11 @@ private func createResolvedPackages(
         // Make sure these module names are unique in the graph.
         let dependencyModuleNames = dependencies.lazy.flatMap{ $0.modules }.map{ $0.name }
         if let duplicateModules = dependencyModuleNames.duplicates(modules.lazy.map{$0.name}) {
-            errors.append(ModuleError.duplicateModule(duplicateModules.first!))
+            engine.emit(ModuleError.duplicateModule(duplicateModules.first!))
```

Comment:
Maybe I'm confused about what this is doing - is the engine "emiting" an error that you're passing through with this call, or are you adding an error to the engine that will cause it to emit some (possibly different) messages as a side effect? Or perhaps more directly - adding the errors *may* cause the engine to emit callbacks or events or whatever, but it's also giving the engine a change to choose what to do with the events being added - so it could translate, localize, provide a collection of errors via state for some external system, etc.